flash/image/detection/data.py

```diff
@@ -95,25 +128,116 @@ def from_coco(
 
+        To learn how to customize the transforms applied for each stage, read our
+        :ref:`customizing transforms guide <customizing_transforms>`.
+
         Args:
-            train_folder: The folder containing the train data.
-            train_ann_file: The COCO format annotation file.
-            val_folder: The folder containing the validation data.
-            val_ann_file: The COCO format annotation file.
-            test_folder: The folder containing the test data.
-            test_ann_file: The COCO format annotation file.
-            predict_folder: The folder containing the predict data.
-            train_transform: The dictionary of transforms to use during training which maps
-                :class:`~flash.core.data.io.input_transform.InputTransform` hook names to callable transforms.
-            val_transform: The dictionary of transforms to use during validation which maps
-                :class:`~flash.core.data.io.input_transform.InputTransform` hook names to callable transforms.
-            test_transform: The dictionary of transforms to use during testing which maps
-                :class:`~flash.core.data.io.input_transform.InputTransform` hook names to callable transforms.
-            predict_transform: The dictionary of transforms to use during predicting which maps
-                :class:`~flash.core.data.io.input_transform.InputTransform` hook names to callable transforms.
-            input_cls: The :class:`~flash.core.data.io.input.Input` used to create the dataset.
-            transform_kwargs: Keyword arguments provided to the transform on instantiation.
-            data_module_kwargs: Keyword arguments provided to the DataModule on instantiation.
+            train_folder: The folder containing images to use when training.
+            train_ann_file: The COCO format annotation file to use when training.
+            val_folder: The folder containing images to use when validating.
+            val_ann_file: The COCO format annotation file to use when validating.
+            test_folder: The folder containing images to use when testing.
+            test_ann_file: The COCO format annotation file to use when testing.
+            predict_folder: The folder containing images to use when predicting.
+            train_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when training.
+            val_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when validating.
+            test_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when testing.
+            predict_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when
+              predicting.
+            input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the data.
+            transform_kwargs: Dict of keyword arguments to be provided when instantiating the transforms.
```

Comment:
Currently it's a bit strange because you can pass a different transform class for train, val, test, predict, but they all share the same keyword arguments. We could think of just passing the instance, like this:
```
datamodule = ...from_x(
    train_transform = ImageInputTransform(image_size=64),
    ...
)
```
It has issues if you want to change e.g. the image size for all transforms but could be an option